spec/utils.tsx

```diff
@@ -1,19 +1,20 @@
 import { ReactNode } from "react"
 import { NostoContext, NostoContextType } from "../src/context"
-import { NostoClient } from "../src/types"
 import { screen, waitFor } from "@testing-library/react"
 import { expect } from "vitest"
+import { nostojs, initNostoStub } from "nosto-js"
+import { API } from "nosto-js/client"
 
 const WAIT_FOR_TIMEOUT = 2000
 
 export function listenTo(event: string) {
   const requests: unknown[] = []
-  if (!window.nostojs) {
-    window.nostojs = (cb: (api: NostoClient) => void) => {
-      (window.nostojs.q = window.nostojs.q || []).push(cb)
-    }
-  }
-  window.nostojs(api => api.listen(event, req => requests.push(req)))
+  initNostoStub()
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-expect-error
+  nostojs(api => api.listen(event as Parameters<API['listen']>[0], (req) => {
```

Comment:
move event type change to top level function?

Reply:
This is not a real PR so far. More of a "here's how you could theoretically do it".

Reply:
I've figured out why there were some issues here. It's now clean 🙂